src/android/Peripheral.java

```diff
@@ -507,8 +528,25 @@ public void onCharacteristicRead(BluetoothGatt gatt, BluetoothGattCharacteristic
             if (readCallback != null) {
                 if (status == BluetoothGatt.GATT_SUCCESS) {
                     readCallback.success(characteristic.getValue());
+                    Bundle bundle = new Bundle();
+                    bundle.putString("DEVICE_NAME", this.device.getName());
+                    bundle.putInt("PERIPHERAL_TYPE", this.device.getType());
+//                    bundle.putInt("READING_TIMEDIFF_OFFSET", );
+                    if (advertisingRSSI != FAKE_PERIPHERAL_RSSI) {
+                        bundle.putString("BT_RSSI", Integer.toString(this.advertisingRSSI));
+                    }                    mFirebaseAnalytics.logEvent(BTAnalyticsLogTypes.BT_READING_SUCCESS.toString(), bundle);
                 } else {
                     readCallback.error("Error reading " + characteristic.getUuid() + " status=" + status);
+                    Bundle bundle = new Bundle();
+                    bundle.putString("DEVICE_NAME", this.device.getName());
+                    bundle.putInt("PERIPHERAL_TYPE", this.device.getType());
+                    bundle.putInt("REASON", status);
+//                    bundle.putInt("READING_TIMEDIFF_OFFSET", );
+                    if (advertisingRSSI != FAKE_PERIPHERAL_RSSI) {
+                        bundle.putString("BT_RSSI", Integer.toString(this.advertisingRSSI));
+                    }
+                    mFirebaseAnalytics.logEvent(BTAnalyticsLogTypes.BT_READING_SUCCESS.toString(), bundle);
```

Comment:
BT_READING_ERROR need to be tracked from Ionic layer after evaluation of different code branches and cases of failure